### PR TITLE
Changes a few children of obj/effect to be children of obj/structure.

### PR DESCRIPTION
**Pull request by phil235**
- Changes obj/effect/blob,spider,swarmer,energy_net,spacevines to be obj/structures instead, since you can hit and destroy them. For future coders: if your object can be hit or damaged, it shouldn't be an obj/effect.
- Adds a large ash and large molten item decal (for future use)
- Simplifies decal/cleanable/replace_decal() code a bit.
- Changes construction() proc name to on_construction() because it's a better fit. Same for deconstruction().
- Made a deconstruct() proc for machinery and computer (also for future use).

### CODE REVIEW

**Comment by phil235** [comment]
This PR is mostly stuff taken from my damage refactor branch. I'm moving it outside so the refactor PR don't get too big.

File: _maps/RandomRuins/LavaRuins/lavaland_surface_pizzaparty.dmm [hunk 1]
```diff
@@ -103,7 +103,7 @@
 	},
 /area/ruin/unpowered)
 "n" = (
-/obj/effect/glowshroom/single,
+/obj/structure/reflector/single,
```
Comment (ChangelingRain) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_pizzaparty.dmm]: This seems wrong.

**Comment by ChangelingRain** [comment]
However, that appears to be the only thing that's wrong here. Good job, phil.

File: _maps/RandomRuins/LavaRuins/lavaland_surface_pizzaparty.dmm (hunk 1 above)
Comment (phil235) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_pizzaparty.dmm]: how the fuck did that happen....

**Pull request merged**